types/sentry.d.ts

```diff
@@ -0,0 +1,39 @@
+import { Options as WebpackOptions } from 'webpack'
+import { Options as SentryOptions } from '@sentry/types'
+import { SentryCliPluginOptions } from '@sentry/webpack-plugin'
+
+export type IntegrationsConfiguration = Record<string, unknown>
+
+export interface LazyConfiguration {
+    chunkName?: string
+    injectLoadHook?: boolean
+    injectMock?: boolean
+    mockApiMethods?: boolean | string[]
+    webpackPrefetch?: boolean
+    webpackPreload?: boolean
+}
+
+export interface ModuleConfiguration {
+    attachCommits?: boolean
+    clientConfig?: SentryOptions
+    clientIntegrations?: IntegrationsConfiguration
+    config?: SentryOptions
+    disableClientRelease?: boolean
+    disableClientSide?: boolean
+    disabled?: boolean
+    disableServerRelease?: boolean
+    disableServerSide?: boolean
+    dsn?: string | ''
+    initialize?: boolean
+    lazy?: boolean | LazyConfiguration
+    logMockCalls?: boolean
+    publishRelease?: boolean
+    repo?: string | ''
+    serverConfig?: SentryOptions
+    serverIntegrations?: IntegrationsConfiguration
+    sourceMapStyle?: WebpackOptions.Devtool
+    webpackConfig?: SentryCliPluginOptions
+}
+
+type LazyConfigurationResolved = Required<LazyConfiguration>
+type ModuleConfigurationResolved = Required<ModuleConfiguration>
```

Comment:
These types are not used ? Maybe you wanted to export them ?

Reply:
Those are internal and I'm using them through JSDoc like so: https://github.com/nuxt-community/sentry-module/blob/bc65423448b391fa54c3322b19331f429e6deeea/lib/core/hooks.js#L30

I suppose I can create them dynamically in JSDoc comments instead.

Reply:
I'm creating those in JSDoc now although still not sure which of the three possible options I should go with 👍 
 - declare internal here and not export
 - declare and export
 - create directly in jsdoc

Reply:
Didn't know it was possible to use unexported types with JSDoc ^^

If only used in internals :
 - create directly in jsdoc
 - declare and export (but not export in main types entry a.k.a. index.d.ts)

Current state looks good to me

Reply:
Thanks